micro/goal.go

```diff
@@ -26,25 +26,24 @@ func EmptyState() *State {
 }
 
 // Substitutions is a list of substitutions represented by a sexprs pair.
-type Substitutions []*Substitution
+type Substitutions map[string]*ast.SExpr
 
 func (s Substitutions) String() string {
-	ss := deriveFmapSs(func(s *Substitution) *ast.SExpr {
-		return ast.Cons(&ast.SExpr{Atom: &ast.Atom{Var: &ast.Variable{Name: s.Var}}}, s.Value)
-	}, []*Substitution(s))
-	l := ast.NewList(ss...).String()
+    sexprs := make([]*ast.SExpr, len(s))
+    ss := map[string]*ast.SExpr(s)
```

Comment:
unfortunately without this cast goderive marked deriveKeys as 'not yet able to generate'.
is that intended or another goderive TODO?

Reply:
No I think that is another bug for goderive :)